packages/fetch-and-compile/lib/types.ts

```diff
@@ -0,0 +1,38 @@
+import type { WorkflowCompileResult } from "@truffle/compile-common";
+import type { SourceInfo } from "@truffle/source-fetcher";
+
+export type FailureType = "fetch" | "compile";
+
+export interface SingleResult {
+  compileResult: WorkflowCompileResult;
+  sourceInfo: SourceInfo;
+}
+
+export interface FetchExternalErrors {
+  fetch: string[]; //addresses
+  compile: string[]; //addresses
+  fetchers: string[]; //fetcher names
+}
+
+export interface Recognizer {
+  getUnrecognizedAddresses(): string[];
+  getAnUnrecognizedAddress(): string | undefined;
```

Comment:
But really... how much is this doing? It seems awkward alongside `getUnrecognizedAddresses`. Maybe it can be named differently to highlight the distinction better? Or maybe one of them can be removed?

Reply:
Well, they're actually not redundant with one another.  Ideally they'd be named to highlight that better.  The difference is that `getUnrecognizedAddresses` gets all unrecognized addresses, whiel `getAnUnrecognizedAddress` specifically gets one that hasn't been marked as unrecognizable.

There's also a different way we could do this, which is to replace `getUnrecognizedAddresses` with `isAddressUnrecognized`.  That's probably the better way in terms of making good abstractions, since that's all we need it for; why return an array to search when all we're going to do is search it?

Yeah I think I'll just do that.